### DISCUSSION TITLE
ActivitySamplingResult.PropagationData overidden if not root span

### DISCUSSION THREAD

**Discussion by verdie-g** [open-telemetry/opentelemetry-dotnet#4687] [General]
Hi, I have an ASP.NET Core app with an endpoint that just writes a message to a Kafka topic. I'm using a `ParentBasedSampler(TraceIdRatioBasedSampler(X))` sampler and I've noticed that whenever the trace is not sampled, the StartActivity will return null for the Kafka span when I would expect a non-recorded activity so I would still be able to propagate the tracing context.

When debugging the OTEL SDK, I've noticed this code
https://github.com/open-telemetry/opentelemetry-dotnet/blob/0f9f507ed6aa960acbcd58e79b9e45330f6432a1/src/OpenTelemetry/Trace/TracerProviderSdk.cs#L423-L472
If the `ActivitySamplingResult` is `PropagationData` but this is not the root span or the parent span is not a remote one then the `ActivitySamplingResult` is replaced by `None`.

I'm not sure how to propagate the tracing context 🤔

Potentially related issue https://github.com/open-telemetry/opentelemetry-dotnet/issues/2887

**Comment by vishweshbankwar** [comment]
@verdie-g Would injecting `Activity.Current` work, instead of injecting the context from specific activity that you are creating? In your case, root activity's context will be injected when the sampling decision is to drop which should be ok if your intention is propagate the sampling decision to downstream service. 

Related: https://github.com/open-telemetry/opentelemetry-dotnet/issues/3290

**Reply by verdie-g** [reply]
I think using `Activity.Current` would propagate the ASP.NET Core span so the kafka consume span would have that span as parent which will be weird when viewing the trace.

**Reply by vishweshbankwar** [reply]
If the decision is to drop then that entire trace will not be exported so, that should not be an issue. If the decision is to record, your StartActivity will return non-null activity, which will be the current and will be propagated.

**Reply by verdie-g** [reply]
Makes sense. Thanks.
